PythonVisualizations/Heap.py

```diff
@@ -0,0 +1,518 @@
+        callEnviron = self.createCallEnvironment()
+        self.startAnimations()
+
+        self.size = val
+        del self.list[:]
+        self.display()
+
+        # adding the values to the list
+        for i in range(val):
+            num = random.randint(1, 99)
+
+            # don't move arrow up if the first cell is being filled because it is already pointing there
+            if len(self.list) >= 1:
+                self.moveItemsBy(self.indexDisplay, (0, + self.CELL_HEIGHT))
+
+            cellCoords = self.cellCoords2(len(self.list))  # Color box
+            cellCenter = self.cellCenter2(len(self.list))  # Number in box
+
+            # create new cell and cell value display objects
+            cellCoords = add_vector(cellCoords, (0, 0, 0, self.CELL_BORDER))
+            toPositions = (cellCoords, cellCenter)
+
+            # determine the top left and bottom right positions
+            startPosition = [self.HEAP_X0, 0, self.HEAP_X0, 0]  # Color
+            startPosition = add_vector(startPosition, (0, 0, self.CELL_WIDTH, self.CELL_HEIGHT))  # color`
+            cellPair = self.createCellValue(startPosition, num)
+            callEnviron.add(cellPair)
+            self.moveItemsTo(cellPair, toPositions, steps=self.CELL_HEIGHT, sleepTime=0.01)
+
+            # add a new Drawable with the new value, color, and display objects
+            d = drawable(
+                num, self.canvas.itemconfigure(cellPair[0], 'fill')[-1], *cellPair)
+            self.list.append(d)  # store item at the end of the list
+
+        # finish the animation
+        self.cleanUp(callEnviron)
+
+
+    def cellCoords(self, cell_index):  # Get bounding rectangle for array cell
+        return (self.HEAP_X0 + self.CELL_BORDER,  # width
+                (self.HEAP_Y0 + self.CELL_HEIGHT * (cell_index + 1)) + self.CELL_BORDER,  # height
+                self.HEAP_X0 + self.CELL_WIDTH - self.CELL_BORDER,
+                self.HEAP_Y0 + self.CELL_HEIGHT * cell_index - self.CELL_BORDER)
+
+
+    def cellCoords2(self, cell_index):  # Get bounding rectangle for array cell
+        return (self.HEAP_X0 + self.CELL_BORDER,  # width
+                (self.HEAP_Y0 + self.CELL_HEIGHT * (cell_index + 1)),  # height
+                self.HEAP_X0 + self.CELL_WIDTH - self.CELL_BORDER,
+                self.HEAP_Y0 + self.CELL_HEIGHT * cell_index)
+
+
+    def cellCenter(self, index):  # Center point for array cell at index
+        half_cell_x = (self.CELL_WIDTH - self.CELL_BORDER) // 2
+        half_cell_y = (self.CELL_HEIGHT - self.CELL_BORDER) // 2
+        return subtract_vector(self.cellCoords(index), (half_cell_x, half_cell_y))
+
+
+    def cellCenter2(self, index):  # Center point for array cell at index
+        half_cell_x = (self.CELL_WIDTH + self.CELL_BORDER) // 2
+        half_cell_y = (self.CELL_HEIGHT - self.CELL_BORDER) // 2
+
+        return add_vector(subtract_vector(self.cellCoords(index), (0, half_cell_y)), (half_cell_x, 0))

```

Comment:
In general, it's a bad idea to have two, nearly identical functions with nearly identical names.  Other developers won't know what the distinction is or when to use one versus the other.  The version of Heap.py in `origin/master` standardized these two into a single function.